### ISSUE TITLE
Panic error for an approle on backend

### ISSUE THREAD

**Issue by bartlomiejdanek** [terraform-providers/terraform-provider-vault#489]
### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.

```
Terraform v0.12.6
+ provider.vault v2.1.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- vault_approle_auth_backend_role

### Terraform Configuration Files
```hcl
resource "vault_auth_backend" "approle" {
  type = "approle"
}

resource "vault_approle_auth_backend_role" "cicd" {
  backend   = vault_auth_backend.approle.path
  role_name = "cicd-role"
  policies  = ["default"]
}
```

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

```
Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is nil, not json.Number
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: goroutine 19 [running]:
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: github.com/terraform-providers/terraform-provider-vault/vault.approleAuthBackendRoleRead(0xc0002ac9a0, 0x1eb1bc0, 0xc0001628c0, 0x1cc2180, 0x2c5fb20)
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:         /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vault/resource_approle_auth_backend_role.go:254 +0x1f8c
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: github.com/terraform-providers/terraform-provider-vault/vault.approleAuthBackendRoleCreate(0xc0002ac9a0, 0x1eb1bc0, 0xc0001628c0, 0x2, 0x2c5fb20)
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:         /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vault/resource_approle_auth_backend_role.go:181 +0xc15
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc000148f00, 0xc0000b6870, 0xc000488820, 0x1eb1bc0, 0xc0001628c0, 0xc0003d6e01, 0xc0004870e0, 0x1d2a9e0)
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:         /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:286 +0x3b4
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc000149700, 0xc0001efa30, 0xc0000b6870, 0xc000488820, 0xc0004870e8, 0xc0000a8028, 0x1d2cae0)
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:         /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:285 +0x18f
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0000ae178, 0x21772e0, 0xc000130e70, 0xc0000b0660, 0xc0000ae178, 0xc000130e70, 0xc00024aba8)
2019-08-05T14:23:19.621+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:         /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/helper/plugin/grpc_provider.go:851 +0x894
2019-08-05T14:23:19.622+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1e8d680, 0xc0000ae178, 0x21772e0, 0xc000130e70, 0xc0000b6410, 0x0, 0x21772e0, 0xc000130e70, 0xc00041c1e0, 0x1dd)
2019-08-05T14:23:19.622+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:         /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/github.com/hashicorp/terraform/internal/tfplugin5/tfplugin5.pb.go:3217 +0x23e
2019-08-05T14:23:19.622+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: github.com/terraform-providers/terraform-provider-vault/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc000552600, 0x2183fe0, 0xc000553200, 0xc00029dc00, 0xc00013f800, 0x2c345a0, 0x0, 0x0, 0x0)
2019-08-05T14:23:19.622+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:         /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/google.golang.org/grpc/server.go:972 +0x470
2019-08-05T14:23:19.622+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: github.com/terraform-providers/terraform-provider-vault/vendor/google.golang.org/grpc.(*Server).handleStream(0xc000552600, 0x2183fe0, 0xc000553200, 0xc00029dc00, 0x0)
2019-08-05T14:23:19.622+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:         /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/google.golang.org/grpc/server.go:1252 +0xda6
2019-08-05T14:23:19.622+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: github.com/terraform-providers/terraform-provider-vault/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000440580, 0xc000552600, 0x2183fe0, 0xc000553200, 0xc00029dc00)
2019-08-05T14:23:19.622+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:         /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/google.golang.org/grpc/server.go:691 +0x9f
2019-08-05T14:23:19.622+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4: created by github.com/terraform-providers/terraform-provider-vault/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
2019-08-05T14:23:19.622+0200 [DEBUG] plugin.terraform-provider-vault_v2.1.0_x4:         /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-vault/vendor/google.golang.org/grpc/server.go:689 +0xa1
2019/08/05 14:23:19 [DEBUG] vault_approle_auth_backend_role.cicd: apply errored, but we're indicating that via the Error pointer rather than returning it: rpc error: code = Unavailable desc = transport is closing
2019/08/05 14:23:19 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2019/08/05 14:23:19 [TRACE] EvalMaybeTainted: vault_approle_auth_backend_role.cicd encountered an error during creation, so it is now marked as tainted
2019/08/05 14:23:19 [TRACE] <root>: eval: *terraform.EvalWriteState
2019/08/05 14:23:19 [TRACE] EvalWriteState: removing state object for vault_approle_auth_backend_role.cicd
2019/08/05 14:23:19 [TRACE] <root>: eval: *terraform.EvalApplyProvisioners
2019/08/05 14:23:19 [TRACE] EvalApplyProvisioners: vault_approle_auth_backend_role.cicd has no state, so skipping provisioners
2019/08/05 14:23:19 [TRACE] <root>: eval: *terraform.EvalMaybeTainted
2019/08/05 14:23:19 [TRACE] EvalMaybeTainted: vault_approle_auth_backend_role.cicd encountered an error during creation, so it is now marked as tainted
2019/08/05 14:23:19 [TRACE] <root>: eval: *terraform.EvalWriteState
2019-08-05T14:23:19.624+0200 [DEBUG] plugin: plugin process exited: path=/Users/bard/projects/vault-config/.terraform/plugins/darwin_amd64/terraform-provider-vault_v2.1.0_x4 pid=28452 error="exit status 2"
2019/08/05 14:23:19 [TRACE] EvalWriteState: removing state object for vault_approle_auth_backend_role.cicd
2019/08/05 14:23:19 [TRACE] <root>: eval: *terraform.EvalIf
2019/08/05 14:23:19 [TRACE] <root>: eval: *terraform.EvalIf
2019/08/05 14:23:19 [TRACE] <root>: eval: *terraform.EvalWriteDiff
2019/08/05 14:23:19 [TRACE] <root>: eval: *terraform.EvalApplyPost
2019/08/05 14:23:19 [ERROR] <root>: eval: *terraform.EvalApplyPost, err: rpc error: code = Unavailable desc = transport is closing
2019/08/05 14:23:19 [ERROR] <root>: eval: *terraform.EvalSequence, err: rpc error: code = Unavailable desc = transport is closing
2019/08/05 14:23:19 [TRACE] [walkApply] Exiting eval tree: vault_approle_auth_backend_role.cicd
2019/08/05 14:23:19 [TRACE] vertex "vault_approle_auth_backend_role.cicd": visit complete
2019/08/05 14:23:19 [TRACE] dag/walk: upstream of "meta.count-boundary (EachMode fixup)" errored, so skipping
2019/08/05 14:23:19 [TRACE] dag/walk: upstream of "provider.vault (close)" errored, so skipping
2019/08/05 14:23:19 [TRACE] dag/walk: upstream of "root" errored, so skipping
2019/08/05 14:23:19 [TRACE] statemgr.Filesystem: not making a backup, because the new snapshot is identical to the old
2019/08/05 14:23:19 [TRACE] statemgr.Filesystem: no state changes since last snapshot
2019/08/05 14:23:19 [TRACE] statemgr.Filesystem: writing snapshot at terraform.tfstate
2019/08/05 14:23:19 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2019/08/05 14:23:19 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock
2019-08-05T14:23:19.639+0200 [DEBUG] plugin: plugin exited
```

### Expected Behavior
create an approle on backend

### Actual Behavior
panic

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by lawliet89** [comment]
What version of Vault are you using?

**Comment by bartlomiejdanek** [comment]
@lawliet89 I use 1.2.0, should I downgrade to 1.1.X?

**Comment by lawliet89** [comment]
There is a breaking change in Vault 1.2. It should be fixed #487. You might need to wait for a new release or use a build from the latest master branch.

**Comment by bartlomiejdanek** [comment]
Ok, thanks! I guess that we can close the issue.
